discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -122,64 +196,19 @@ def format_disk(cls, disk_path):
 
     @staticmethod
     def _get_all_scsi_disks(node):
-        """
-        :return: All node disks that use an SCSI bus (/dev/sd*)
-        """
-
-        def is_scsi_disk(disk):
-            return any(target.getAttribute('bus') == 'scsi' for target in
-                       disk.getElementsByTagName('target'))
-
-        all_disks = minidom.parseString(node.XMLDesc()).getElementsByTagName('disk')
-
-        return [disk for disk in all_disks if is_scsi_disk(disk)]
-
-    @staticmethod
-    def _get_disk_source_file(disk):
-        sources = disk.getElementsByTagName('source')
-
-        assert len(sources) in (0, 1), f"A disk must have either 0 or 1 sources, {sources}"
-
-        if len(sources) == 0:
-            return None
-
-        return sources[0].getAttribute('file')
-
-    @staticmethod
-    def _get_disk_alias(disk):
-        aliases = disk.getElementsByTagName('alias')
-
-        assert len(aliases) in (0, 1), f"A disk must have either 0 or 1 aliases, {aliases}"
-
-        if len(aliases) == 0:
-            return None
-
-        return aliases[0].getAttribute('name')
+        return filter(lambda disk: disk.bus == 'scsi', LibvirtController._list_disks(node))
```

Comment:
Consider using generator expressions with `if` rather than `filter(lambda`:

`return (disk for disk in LibvirtController._list_disks(node) if disk.bus == 'scsi')`